pinot-query-planner/src/main/java/org/apache/pinot/query/parser/CalciteRexExpressionParser.java

```diff
@@ -64,12 +65,34 @@ public static List<Expression> addSelectList(List<Expression> existingList, List
     final Iterator<RexExpression> iterator = rexNodeList.iterator();
     while (iterator.hasNext()) {
       final RexExpression next = iterator.next();
-      selectExpr.add(toExpression(next, pinotQuery));
+      if (next.getKind() == SqlKind.WINDOW) {
+        RexExpression.WindowFunctionExpression window = (RexExpression.WindowFunctionExpression) next;
+        // first add all columns from aggregate function
+        addColumnsToSelectList(selectExpr, window.getFunctionCall().getFunctionOperands(), pinotQuery);
+
+        // add columns from partition list
+        addColumnsToSelectList(selectExpr, window.getPartitionKeys(), pinotQuery);
+
+        // add columns from order list
+        addColumnsToSelectList(selectExpr, window.getOrderKeys(), pinotQuery);
+      } else {
+        selectExpr.add(toExpression(next, pinotQuery));
+      }
     }
```

Comment:
why would a WINDOW node be parsed in the `addSelectList` util function? you should create a parse WINDOW util function instead